### PR TITLE
89477 fix link text icon

### PR DESCRIPTION
**Pull request by jzucadi**
## Summary

When a link text has a chevron svg at the end of it, the svg icon gets rendered on the next line on certain screen sizes in between desktop and phone. 

This also occurs on the alert expandable component on the landing page. This PR fixes this svg issue so that the the svg icon is accompanied by part of the sentence text and is not rendered on it's own on the next line.

## Related issue(s)

department-of-veterans-affairs/va.gov-team#89477

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |   Screenshot 2024-07-30 at 4 45 13 PM | Screenshot 2024-08-14 at 1 55 30 PM |
| Desktop |   Screenshot 2024-07-30 at 4 45 13 PM     |   Screenshot 2024-08-13 at 5 08 39 PM
 |

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  | Screenshot 2024-08-13 at 5 01 39 PM |  Screenshot 2024-08-13 at 5 01 19 PM |
| Desktop  | Screenshot 2024-08-13 at 5 01 39 PM |  Screenshot 2024-08-13 at 5 01 19 PM |

## What areas of the site does it impact?

MHV landing page

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by dcloud — COMMENTED** [review]
It occurred to me that some new-ish CSS, specifically `text-wrap: balance;`, might be a nice solution, though its a [recent-ish addition](https://caniuse.com/css-text-wrap-balance), so a [polyfill](https://github.com/adobe/balance-text) might be needed.

**Comment by fmccaf1** [comment]
Hey @dcloud and @jzucadi, this is close but not quite spot-on. I created this below to clarify the difference for the desktop cards vs. mobile: 

image

If you want to look at it in Figma, [it's in this file](https://www.figma.com/design/CAChU51fWYMZsgDR5RXeSc/MHV-Landing-Page?node-id=1727-33108&t=zt9XXgt6TIaU5kbf-1)

**Comment by fmccaf1** [comment]
Oh one other thing, I think you might be doing this but just want to call it out in case. The blue alert text also has this problem (caret falls off the line by itself):

image

**Comment by Andrew565** [comment]
Contains failing tests, choosing not to review at this time.
Andrew Steele, T2 Support

**Review by rmessina1010 — DISMISSED** [review]
LGTM

**Review by radavis — COMMENTED** [review]
The Desktop changes look good. The Mobile changes don't align with what @fmccaf1 has requested (chevron being right-aligned). 

mobile view:
Screenshot 2024-08-14 at 09-37-14 My HealtheVet Veterans Affairs

**Review by carlosfelix2 — DISMISSED** [review]

File: src/applications/mhv-landing-page/components/NavCard.jsx [hunk 1]
```diff
@@ -30,15 +30,16 @@ const NavCard = ({
...
-          {text} {isExternal && externalLinkText}
+          {text}
+          {isExternal && ` ${externalLinkText}`}
+          {!isExternal && (
+            <va-icon
+              class="vads-u-margin-right--neg1 medium-screen:vads-u-margin-right--0"
+              icon="navigate_next"
+              size={4}
+            />
+          )}
```
Comment (dcloud) [inline src/applications/mhv-landing-page/components/NavCard.jsx:41]: I think by moving the `<va-icon>` within the `<span>`, it breaks the small-screen layout where the chevron is right-aligned

**Review by dcloud — COMMENTED** [review]
I'm with @radavis that currently the mobile-specific layout is broken. I think this will be good once that is restored

**Review by dcloud — COMMENTED** [review]
I've restarted this review a few times because this was unexpectedly hairy. I think you were going for a fix for browsers that don't support `text-wrap: balance;`, but if I comment that bit of CSS out, things break.

I also think `.desktop-icon` was supposed to be `.mhv-c-link-icon-desktop`, and the same for the mobile one?

I think we may need to be okay with a fix for latest browsers, or the ability to test in older browsers, since text layout is such a finicky thing on the web.

**Comment by jzucadi** [comment]
> I've restarted this review a few times because this was unexpectedly hairy. I think you were going for a fix for browsers that don't support `text-wrap: balance;`, but if I comment that bit of CSS out, things break.
> 
> I also think `.desktop-icon` was supposed to be `.mhv-c-link-icon-desktop`, and the same for the mobile one?
> 
> I think we may need to be okay with a fix for latest browsers, or the ability to test in older browsers, since text layout is such a finicky thing on the web.

You are right, I did forget to update those css class names for the breakpoint. For better support for older browsers I originally thought `word-break: break-word` would be enough for older browsers but I have now added in more properties to help cover this.

**Review by dcloud — COMMENTED** [review]
**I think we should ship the tiny `text-wrap: balance;` and pursue alternative solutions like giving tablet screens the mobile layout.**

Still looks good when `text-wrap: balance` is enabled, but I don't think any of the other changes have an effect. 

When I view this page in iOS 17.0.1 (21A342) Simulator, which would appears to be running Safari 17.0, I see the hanging chevron:

Simulator Screenshot - iPad (10th generation) - 2024-08-15 at 17 25 57

**Comment by dcloud** [comment]
Per [Google](https://developer.chrome.com/docs/css-ui/css-text-wrap-balance):

> Without text-wrap: balance; designers, content editors and publishers have [few tools](https://developer.mozilla.org/docs/Web/CSS/CSS_Text/Wrapping_Text) to change the way lines are balanced.. The best options available being to use [<wbr>](https://developer.mozilla.org/docs/Web/HTML/Element/wbr) or [&shy;](https://developer.mozilla.org/docs/Web/CSS/hyphens) to help guide text layouts into smarter decisions about where to break lines and words.

The tools that are mentioned focus on wrapping long strings of text that ordinarily wouldn't break, whereas this scenario involves a place where there is a natural break between "words", though it looks displeasing to people.

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Pull request closed without merging**